guides/common/modules/proc_installing-and-configuring-puppet-agent-during-host-registration.adoc

```diff
@@ -0,0 +1,63 @@
+* You created an activation key that enables the *{project-client-name}* repository for hosts.
+For more information, see {ContentManagementDocURL}Managing_Activation_Keys_content-management[Managing Activation Keys] in _{ContentManagementDocTitle}_.
+endif::[]
+ifdef::katello,orcharhino[]
+.Prerequisites
+* You created a Product and repository for the upstream Puppet agent, such as `\https://yum.puppet.com` or `\https://apt.puppet.com`, and synchronized the repository to {Project}.
+For more information, see {ContentManagementDocURL}Importing_Content_content-management[Importing Content] in _{ContentManagementDocTitle}_.
+* You created an activation key that enables the Puppet agent repository for hosts.
+For more information, see {ContentManagementDocURL}Managing_Activation_Keys_content-management[Managing Activation Keys] in _{ContentManagementDocTitle}_.
+endif::[]
+
+.Procedure
+. In the {ProjectWebUI}, navigate to *Configure* > *Global Parameters* to add host parameters globally.
+Alternatively, you can navigate to *Configure* > *Host Groups* and edit or create a host group to add host parameters only in a host group.
+ifdef::foreman-el,foreman-deb[]
+. Enable the Puppet agent from the official Puppet repository using one of the following host parameters in global parameters or a host group:
+
+* Add a host parameter named `enable-puppetlabs-repo` for the latest stable Puppet release in the unversioned repository.
+* Add a host parameter named `enable-official-puppet7-repo` for the Puppet 7 release repository.
+* Add a host parameter named `enable-puppetlabs-puppet6-repo` for the Puppet 6 release repository.
+
++
+Select the *boolean* type and set the value to `true`.
+endif::[]
+ifdef::katello,orcharhino,satellite[]
+. Enable the Puppet agent using a host parameter in global parameters or a host group.
+endif::[]
+ifdef::katello,orcharhino[]
+Add a host parameter named `enable-puppet7` for Puppet 7 or `enable-puppet6` for Puppet 6, select the *boolean* type, and set the value to `true`.
+endif::[]
+ifdef::satellite[]
+Add a host parameter named `enable-puppet7` for Puppet 7, select the *boolean* type, and set the value to `true`.
+endif::[]
+. Specify configuration for the Puppet agent using all the following host parameters in global parameters or a host group:
+* Add a host parameter named `puppet_server`, select the *string* type, and set the value to the hostname of your {ProjectServer} or any {SmartProxyServer} that you use as the Puppet server, such as `{foreman-example-com}`.
+You can also use this parameter to configure the Puppet agent for a non-integrated Puppet server.
+* Add a host parameter named `puppet_ca_server`, select the *string* type, and set the value to the hostname of your {ProjectServer} or any {SmartProxyServer} that you use as the Puppet CA server, such as `{foreman-example-com}`.
```

Comment:
You can make this an optional step. If `puppet_ca_server` is not set, Puppet will use the same value as `puppet_server`.

The use case is as follows:
* Puppetserver is configured as CA on `puppet-ca.example.com`. This will only be the CA server
* Puppetserver is configured as compile server on `puppet.example.com` or even `puppet.site-a.example.com`.

On a technical level the puppetserver installs are pretty much the same, except for some configuration changes. In large installations it's common to have multiple compile servers behind a load balancer.

In the simple use case of a single server it's not needed to specify them both, but it's not wrong to be explicit.

I'm not sure if being explicit in both, or making the CA step conditional on it being another server is clearest for the reader, but please do what you think is clearest for users.